src/opencl_rawsha512_fmt_plug.c

```diff
@@ -306,21 +306,22 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 
 	global_work_size = GET_NEXT_MULTIPLE(count, local_work_size);
 
-	///Copy data to GPU memory
-	if (sha512_key_changed || ocl_autotune_running) {
+	// Copy data to GPU memory
+	if (new_keys) {
 		BENCH_CLERROR(clEnqueueWriteBuffer
 		    (queue[gpu_id], mem_in, CL_FALSE, 0, insize, gkey, 0, NULL,
 			multi_profilingEvent[0]), "Copy memin");
+
+		new_keys = 0;
 	}
 
-	///Run kernel
+	// Run kernel
 	BENCH_CLERROR(clEnqueueNDRangeKernel
 	    (queue[gpu_id], crypt_kernel, 1, NULL, &global_work_size, lws,
 		0, NULL, multi_profilingEvent[1]), "Set ND range");
 
-	/// Reset key to unchanged and hashes uncopy to host
-	sha512_key_changed = 0;
-    hash_copy_back = 0;
+	// Reset hashes uncopy to host [SIC]
+	hash_copy_back = 0;
```

Comment:
I don't understand what is meant by "// Reset hashes uncopy to host [SIC]", so this comment is unhelpful - it should either be dropped or edited so that it'd clarify things.